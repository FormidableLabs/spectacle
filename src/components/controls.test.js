import React from 'react';
import { mount } from 'enzyme';
import Controls from './controls';

describe('<Controls />', () => {
  test('should render correctly.', () => {
    const context = {
      styles: {
        controls: {
          prev: { background: '#f00' },
          next: { background: '#f0f' }
        }
      }
    };
    const wrapper = mount(
      <Controls
        currentSlideIndex={2}
        totalSlides={5}
        onPrev={() => {}}
        onNext={() => {}}
      />,
      { context }
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('should call the next function when the next button is selected', () => {
    const context = { styles: { controls: {} } };
    const nextFunc = jest.fn();
    const wrapper = mount(
      <Controls
        currentSlideIndex={2}
        totalSlides={5}
        onPrev={() => {}}
        onNext={nextFunc}
      />,
      { context }
    );
    wrapper
      .findWhere(node => node.name() === 'button' && node.key() === 'next')
      .simulate('click');
    expect(nextFunc).toHaveBeenCalledTimes(1);
  });

  test('should call the prev function when the previous button is selected', () => {
    const context = { styles: { controls: {} } };
    const prevFunc = jest.fn();
    const wrapper = mount(
      <Controls
        currentSlideIndex={3}
        totalSlides={5}
        onPrev={prevFunc}
        onNext={() => {}}
      />,
      { context }
    );
    wrapper
      .findWhere(node => node.name() === 'button' && node.key() === 'prev')
      .simulate('click');
    expect(prevFunc).toHaveBeenCalledTimes(1);
  });
});
