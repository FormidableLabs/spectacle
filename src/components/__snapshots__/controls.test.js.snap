exports[`<Controls /> should render correctly. 1`] = `
<Controls
  currentSlideIndex={2}
  onNext={[Function]}
  onPrev={[Function]}
  totalSlides={5}>
  <div
    data-radium={true}>
    <button
      data-radium={true}
      onClick={[Function]}
      style={
        Object {
          "background": "#f00",
        }
      }
      type="button">
      <svg
        height="32px"
        viewBox="0 0 512 828.586"
        width="32px">
        <path
          d="M512,97.707L414.293,0L0,414.293l414.293,414.293L512,730.88L195.414,414.293L512,97.707z" />
      </svg>
    </button>
    <button
      data-radium={true}
      onClick={[Function]}
      style={
        Object {
          "background": "#f0f",
        }
      }
      type="button">
      <svg
        height="32px"
        viewBox="0 0 512 828.586"
        width="32px">
        <path
          d="M97.707,0L0,97.707l316.586,316.586L0,730.88l97.707,97.706L512,414.293L97.707,0z" />
      </svg>
    </button>
  </div>
</Controls>
`;
